packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -251,7 +259,10 @@ class DOProviderContainerRuntimeFactory extends BaseContainerRuntimeFactory {
 		};
 		super({
 			registryEntries: [rootDataObjectFactory.registryEntry],
-			runtimeOptions: compatibilityModeRuntimeOptions[compatibilityMode],
+			runtimeOptions: {
+				...compatibilityModeRuntimeOptions[compatibilityMode],
+				...runtimeOptionOverrides,
+			},
 			provideEntryPoint,
 			minVersionForCollab: compatibilityModeToMinVersionForCollab[compatibilityMode],
```

Comment:
I think we may need to also override `minVersionForCollab` as well. I have a pending PR that will throw if we set a config to a value that requires a higher `minVersionForCollab` than provided (https://github.com/microsoft/FluidFramework/pull/24625). In this scenario, if a customer sets `createBlobPayloadPending: true`, then it would throw unless `minVersionForCollab` is at least `"2.40.0"` (release when createBlobPayloadPending was added).

Reply:
It may be worth considering to add a `"2.40"` compat mode. We could either use that to set `createBlobPayloadPending: true` automatically in the `compatibilityModeRuntimeOptions` map, or require using `"2.40"` compat mode to be allowed to set `createBlobPayloadPending: true` in the runtime options override object.

Reply:
I think we'd modify the compat modes when createBlobPayloadPending becomes on-by-default (i.e. then it would be set to false for compat modes "1" and "2").  I don't think we'd need to set it to true in any compat mode just because we don't need to be particularly aggressive about enabling it (esp. if the individual customer has this option of enabling it themselves).